### PR TITLE
TST: add test_series_any_timedelta for GH17667

### LINKED ISSUE #17667: DataFrame.any(axis={0, 1}) returns inconsistent values for non-zero timedelta
#### Steps to reproduce

```python
In [1]: import pandas

In [2]: df = pandas.DataFrame({'a': pandas.Series([0, 0]),
   ...:                        't': pandas.Series([pandas.to_timedelta(0, 's'), pandas.to_timedelta(1, 'ms')])})

In [3]: df
Out[3]:
   a               t
0  0        00:00:00
1  0 00:00:00.001000

In [4]: df.dtypes
Out[4]:
a              int64
t    timedelta64[ns]
dtype: object

In [5]: df.any(axis=0)
Out[5]:
a    False
t     True
dtype: bool

In [6]: df.any(axis=1)
Out[6]:
0    False
1    False
dtype: bool
```

#### Problem description

pandas.DataFrame.any() returns "whether any element is True over the selected axis".

In the example above, the results of `df.any(axis=ax)` is clearly inconsistent for the two axis directions (cells [5] and [6]). 

If an element in column `"t"` leads to a result of `True` in `df.any(axis=0)`, then the same element should lead to a result of `True` in `df.any(axis=1)`.

#### Expected output

As a user, I expected `DataFrame.any()` to return True for rows or columns with a _non-zero_ timedelta value.  That is, I expected a `timedelta` column to have the same semantics as integers: zero values are treated as False, while other values are treated as True.

In other words, this is the output I expected:

```
In [6]: df.any(axis=1)
Out[6]:
0    False
1    True
dtype: bool
```

#### Output of ``pd.show_versions()``

<details>

```
In [7]: pandas.show_versions()

INSTALLED VERSIONS
------------------
commit: None
python: 3.4.5.final.0
python-bits: 64
OS: Windows
OS-release: 7
machine: AMD64
processor: Intel64 Family 6 Model 60 Stepping 3, GenuineIntel
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: None.None

pandas: 0.19.1
nose: 1.3.7
pip: 9.0.1
setuptools: 27.2.0
Cython: 0.24.1
numpy: 1.11.2
scipy: 0.18.1
statsmodels: 0.6.1
xarray: 0.8.2
IPython: 5.1.0
sphinx: 1.4.8
patsy: 0.4.1
dateutil: 2.6.0
pytz: 2016.7
blosc: 1.5.0
bottleneck: 1.2.0
tables: 3.2.2
numexpr: 2.6.1
matplotlib: 1.5.1
openpyxl: 2.4.0
xlrd: 1.0.0
xlwt: 1.1.2
xlsxwriter: 0.9.3
lxml: 3.6.4
bs4: 4.5.3
html5lib: 0.999
httplib2: 0.9.2
apiclient: None
sqlalchemy: 1.1.3
pymysql: None
psycopg2: 2.6.2 (dt dec pq3 ext lo64)
jinja2: 2.8
boto: 2.43.0
pandas_datareader: None
```

</details>

### PR DESCRIPTION
**Pull request by rohitsanj**
Test case example is the same as the one given in the issue #17667 

- [x] closes #17667 
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/test_multilevel.py [hunk 1]
```diff
@@ -1065,6 +1065,22 @@ def test_frame_any_all_group(self):
         ex = DataFrame({"data": [False, False]}, index=["one", "two"])
         tm.assert_frame_equal(result, ex)
 
+    def test_series_any_timedelta(self):
```
Comment (simonjayhawkins) [inline pandas/tests/test_multilevel.py]: can you add the issue number

File: pandas/tests/test_multilevel.py [hunk 2]
```diff
@@ -1065,6 +1065,22 @@ def test_frame_any_all_group(self):
...
+        df = DataFrame(
+            {
+                "a": Series([0, 0]),
+                "t": Series([pd.to_timedelta(0, "s"), pd.to_timedelta(1, "ms")]),
+            }
+        )
+
+        result = df.any(axis=0)
+        ex = Series(data=[False, True], index=["a", "t"])
```
Comment (simonjayhawkins) [inline pandas/tests/test_multilevel.py]: use expected

File: pandas/tests/test_multilevel.py [hunk 3]
```diff
@@ -1065,6 +1065,22 @@ def test_frame_any_all_group(self):
...
+        tm.assert_series_equal(result, ex)
+
+        result = df.any(axis=1)
+        ex = Series(data=[False, True])
```
Comment (simonjayhawkins) [inline pandas/tests/test_multilevel.py]: same

**Review by simonjayhawkins — CHANGES_REQUESTED** [review]
@rohitsanj thanks for the PR. couple of suggestions.

File: pandas/tests/test_multilevel.py (hunk 1 above)
Comment (rohitsanj) [reply pandas/tests/test_multilevel.py]: added

**Review by mroeschke — APPROVED** [review]

**Review by simonjayhawkins — APPROVED** [review]

**Comment by simonjayhawkins** [comment]
Thanks @rohitsanj

**Comment by mroeschke** [comment]
@rohitsanj feel free to contribute more tests like this! https://github.com/pandas-dev/pandas/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22+label%3A%22Needs+Tests%22

**Pull request merged**